src/bin/main.rs

```diff
@@ -58,33 +61,70 @@ async fn main() -> Result<(), Error> {
     Ok(())
 }
 
+async fn connect<F>(
+    mut wallet: Wallet<F>,
+    settings: &Settings,
+    status: fn(&str),
+) -> Wallet<F>
+where
+    F: SecureWalletFile + std::fmt::Debug,
+{
+    let con = match (&settings.state.method(), &settings.prover.method()) {
+        (TransportMethod::Tcp, TransportMethod::Tcp) => {
+            wallet
+                .connect_with_status(
+                    TransportTCP::new(&settings.state, &settings.prover),
+                    status,
+                )
+                .await
+        }
+        #[cfg(not(windows))]
+        (TransportMethod::Uds, _) => {
+            wallet
+                .connect_with_status(TransportUDS::new(&settings.state), status)
+                .await
+        }
+
+        (_, _) => panic!("IPC method not supported"),
+    };
+
+    // check for connection errors
+    if con.is_err() {
+        warn!("Connection to Rusk Failed, some operations won't be available.");
+    }
+
+    wallet
+}
+
 async fn exec() -> Result<(), Error> {
     // parse user args
     let args = WalletArgs::parse();
+    // get the subcommand, if any
     let cmd = args.command.clone();
 
     // set symbols to ASCII for Windows terminal compatibility
     #[cfg(windows)]
     requestty::symbols::set(requestty::symbols::ASCII);
 
-    // data directory needs to be clear from the start
-    let data_dir = args
-        .data_dir
-        .as_ref()
-        .cloned()
-        .unwrap_or_else(WalletPath::default_dir);
+    // Get the initial settings from the args
+    let settings = Settings::args(args);
 
-    // create directories
-    fs::create_dir_all(&data_dir)?;
+    // Obtain the profile dir from the settings
+    let profile_folder = settings.profile().clone();
 
-    // set cache directory straight away
-    WalletPath::set_cache_dir(&data_dir)?;
+    fs::create_dir_all(&profile_folder)?;
+
+    // prepare wallet path
+    let wallet_path = WalletPath::from(profile_folder.join("wallet.dat"));
 
     // load configuration (or use default)
-    let mut cfg = Config::load(data_dir)?;
+    let cfg = Config::load(&profile_folder)?;
 
-    // merge static config with parsed args
-    cfg.merge(args);
+    // Finally complete the settings by set the network
+    let settings = settings.network(cfg.network);
+
+    // set cache directory straight away
+    WalletPath::set_cache_dir(&profile_folder)?;
```

Comment:
Why are we not push the network into the profile folder for the cache?
Do you want to handle it in a different PR?

Reply:
Yes, there is no change in this PR related to the cache and it's intentional.
The cache needs to be rewritten by scratch, but in order to address that we need to have the one single wallet PR (this one) first.
The follow up PR would be about the cache, and it will solve both #56 and #58.